src/main/host/syscall/type_formatting.rs

```diff
@@ -224,6 +224,51 @@ simple_debug_impl!(nix::sys::mman::ProtFlags);
 simple_debug_impl!(nix::sys::mman::MapFlags);
 simple_debug_impl!(nix::sys::mman::MRemapFlags);
 
+fn fmt_buffer(
+    f: &mut std::fmt::Formatter<'_>,
+    ptr: PluginPtr,
+    len: usize,
+    options: FmtOptions,
+    mem: &MemoryManager,
+) -> std::fmt::Result {
+    const DISPLAY_LEN: usize = 40;
+
+    if options == FmtOptions::Deterministic {
+        return write!(f, "<pointer>");
+    }
+
+    // read up to one extra character to check if it's a null byte
+    let mem_ref = match mem.memory_ref_prefix(TypedPluginPtr::new::<u8>(ptr, len)) {
+        Ok(x) => x,
+        // the pointer didn't reference any valid memory
+        Err(_) => return write!(f, "{ptr:p}"),
+    };
+
+    let mut s = Vec::with_capacity(DISPLAY_LEN);
```

Comment:
Nit - I think you could use a `String` here instead of later converting from a `Vec`? If so, I think that'd be slightly clearer.

Reply:
I tried this, but the problem is that `std::ascii::escape_default` returns a `u8` iterator, and `String::extend` doesn't exist (a similar replacement is `String::push_str`). So for each call to `escape_default` I think you need to collect the iterator into a new allocated string, and push that to the main string. There could be a better way that I'm missing, but I for now I'll just keep it as a `Vec` to avoid the extra allocations.

Reply:
You can push a character at a time, but yeah it is a little awkward. Up to you.

e.g.
```
use std::convert::TryFrom;

pub fn main() {
    let mut s = String::new();

    let bytes = [0u8; 10];

    for b in bytes {
        for b in std::ascii::escape_default(b) {
            s.push(char::try_from(b as u32).unwrap())
        }
    }

    println!("{}", s);
}
```

Reply:
Okay that seems good to me. Added (using `char::from(u8)` rather than `char::try_from(u32)`).